File: backend/edge-common/src/main/java/io/featurehub/edge/FeatureTransformer.java
package io.featurehub.edge;

import io.featurehub.dacha.model.CacheEnvironmentFeature;
import io.featurehub.dacha.model.CacheFeatureValue;
import io.featurehub.edge.strategies.ClientContext;
import io.featurehub.mr.model.FeatureValueCacheItem;
import io.featurehub.sse.model.FeatureState;

import java.util.List;

public interface FeatureTransformer {
  List<FeatureState> transform(List<CacheEnvironmentFeature> features, ClientContext clientAttributes);
  FeatureState transform(CacheEnvironmentFeature rf, ClientContext clientAttributes);
}

File: backend/edge-common/src/main/kotlin/io/featurehub/edge/FeatureTransformer.kt
package io.featurehub.edge

import io.featurehub.dacha.model.CacheEnvironmentFeature
import io.featurehub.edge.strategies.ClientContext
import io.featurehub.sse.model.FeatureState

interface FeatureTransformer {
  fun transform(features: List<CacheEnvironmentFeature>?, clientAttributes: ClientContext?, allowExtendedProperties: Boolean): List<FeatureState>
  fun transform(rf: CacheEnvironmentFeature, clientAttr: ClientContext?, allowExtendedProperties: Boolean): FeatureState
}
